### DISCUSSION TITLE
ApexDoc with override methods

### DISCUSSION THREAD

**Discussion by pabloagomezmck** [pmd/pmd#4837] [Q&A]
Hi, we have a question about the rule ApexDoc, we don't know if this rule has to notify us when we have an override method.
We think it has to notify us because we need to know the difference beetween the parent method and the child method.
We don't want to report it as a bug without knowing well if it's a bug.
<img width="898" alt="override" src="https://github.com/pmd/pmd/assets/159025372/42581c99-730d-48eb-bf20-a99ba6ea9523">

**Comment by adangel** [comment]
Hi @pabloagomezmck ,
That's currently, how the rule works: from https://docs.pmd-code.org/latest/pmd_rules_apex_documentation.html

> Method overrides and tests are both exempted from having ApexDoc.

So, it's not a bug, it works as designed :smile: But the question remains: Is this correct?

I don't know Apex that well and I don't know, if it supports something like Java's `{@inheritDoc}`...
I guess, it depends on the project's rules, whether you want doc comments for overridden methods as well. And that's why I think, it makes sense to have this feature as an opt-in via a rule property.

HTH,
Andreas
